midp/gfx.js

```diff
@@ -228,20 +230,19 @@ var currentlyFocusedTextEditor;
         }
     }
 
-    function initImageData(imageData, width, height) {
+    function initImageData(imageData, width, height, isMutable) {
         var canvas = document.createElement("canvas");
-        imageData.context = canvas.getContext("2d");
-        setImageDataDimensions(imageData, width, height);
-        return imageData.context;
-    }
-
-    function setImageDataDimensions(imageData, width, height) {
-        var canvas = imageData.context.canvas;
         canvas.width = width;
         canvas.height = height;
 
+        imageData.contextInfo = new ContextInfo(canvas.getContext("2d"));
```

Comment:
Since *contextInfo* is never accessed by Java code, this seems reasonable. I only wonder about the cost of changing the shape of *imageData*.